contracts/sg-ics721/src/execute.rs

```diff
@@ -113,4 +115,70 @@ impl Ics721Execute for SgIcs721Contract {
 
         to_json_binary(&instantiate_msg)
     }
+
+    fn mint_msg(
+        &self,
+        token_id: String,
+        token_uri: Option<String>,
+        owner: String,
+        data: Option<Binary>,
+    ) -> StdResult<Binary> {
+        // parse token data and check whether it is of type NftExtension
+        let extension: Metadata = match data {
+            Some(data) => {
+                match from_json::<NftExtension>(data).ok().map(|ext| Metadata {
```

Comment:
Since everything here is being added from .ext I wonder if we could put this mapping into a helper function

Reply:
No, we cant, sg-ics721 requires struct of type `Metadata`, whilst ics721 requires `NftExtensionMsg`. See comments below.